lib/std/hash.zig

```diff
@@ -37,20 +37,88 @@ pub const XxHash3 = xxhash.XxHash3;
 pub const XxHash64 = xxhash.XxHash64;
 pub const XxHash32 = xxhash.XxHash32;
 
-/// This is handy if you have a u32 and want a u32 and don't want to take a
-/// detour through many layers of abstraction elsewhere in the std.hash
-/// namespace.
-/// Copied from https://nullprogram.com/blog/2018/07/31/
+/// Easy & fast hash function for integer types
+pub fn int(input: anytype) @TypeOf(input) {
+    // This function is only intended for integer types
+    const input_type = @typeInfo(@TypeOf(input));
+    if (input_type != .int) @compileError("std.hash.int only works on integer types.");
```

Comment:
```suggestion
    const info = @typeInfo(@TypeOf(input)).int;
```

I did this already in 5d790e911fbca8bd04392bddad500b96a1837d9c

Reply:
okay done.
is this preferred because a good error message if type != int at some point will be automatically printed?

Reply:
Perhaps the compiler will get "smart" enough to emit a more specific error message here, or perhaps not; but the standard library shouldn't be littered with manual type checks / explicit compile errors when the error you'd otherwise get is relatively understandable, as this one is. Generally speaking, `@compileError` shouldn't be used too often.

Reply:
ok thank you!